service/node.go

```diff
@@ -1576,7 +1576,13 @@ func (s *service) nodeStartup(ctx context.Context) error {
 		log.Debug("vmHost created successfully")
 	}
 
-	symmetrixIDs = s.filterArraysByZoneInfo(s.opts.StorageArrays)
+	// Get the symmetrix ID list from Secret
+	if s.opts.StorageArrays != nil {
```

Comment:
A better check is len(s.opts.StorageArrays) which handles nil and empty values.

Reply:
Agreed!